create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java

```diff
@@ -90,9 +89,18 @@ private boolean isFindable(String user,
                                File file,
                                Publication publication,
                                RequestInfo requestInfo) {
-        return publication.getResourceOwner().getOwner().getValue().equals(user)
-                || requestInfo.userIsAuthorized(EDIT_OWN_INSTITUTION_RESOURCES.toString())
-                || PublicationStatus.PUBLISHED.equals(publication.getStatus()) && file.isVisibleForNonOwner();
+
+        var isEmbargoReader = requestInfo.userIsAuthorized(AccessRight.PUBLISH_THESIS_EMBARGO_READ.toString());
+        var isOwner = publication.getResourceOwner().getOwner().getValue().equals(user);
+
+        if (!file.fileDoesNotHaveActiveEmbargo()) {
```

Comment:
Maybe create method fileHasActiveEmbargo(file)

Reply:
I have refactored to make it super clear what is going on.

however I will not create more functions to hide logic, it is important that it is crystal clear what it being tested here.

Reply:
compiler will simplify local var, but function calls goes to the stack
and you don't know what is being tested

Reply:
embargo has to be tested independently, as a true/false response is only valid IF an embargo IS in place.
otherwise it should be ignored.

Reply:
How many things are you doing in this method right now?

Reply:
I agree with Kir. How many things do you do in this method?

Reply:
You are abstracting to a very high level, I have seen function that calls another function 4 or 5 level deep, without adding any meaningful work to anything.

It has to be two separate tests, we should only test embargoRead on thesis.

The design of the access right system we have now, adds loads of logic into each and every lambda we have. Even worse is the fact that we don't have a uniform way of testing access rights. Maybe it is implemented, or not, impossible to say before you have read all lines of code.
There are only two sensible places to test access, first line of lambda or in the service that read/persist the entities in question. Since aws can do simple access control, it make sense to put our access control into each and every service we got, by creating/extending a service base class that has an abstract hasAccess(action,entity) that needs to be implemented

Here is the old code, which is fine by me, but I made it more explicit.
```
return publication.getResourceOwner().getOwner().getValue().equals(user)
                || requestInfo.userIsAuthorized(EDIT_OWN_INSTITUTION_RESOURCES.toString())
                || PublicationStatus.PUBLISHED.equals(publication.getStatus()) && file.isVisibleForNonOwner();
```
Hiding the logic in functions, you will have to jump into each and every function to make sure it does what it says it does, that makes the code harder to read, not easier. In the code above you need to check `isVisibleForNonOwner` in order to know what circumstance it applies to.

Reply:
Creating readable code is a team effort, and something the team should agree as an whole. The fact that you think the code is more readable as-is, isn't much of an argument if two other team members disagree.

This isn't following clean-code principles, which the rest of the team uses. If we should ditch the clean code principle, and do something else, the team should agree to the change as a unit.

Using function calls directly isn't a problem if you use an IDE and functions are named properly. (Tip: if you press ctrl +left click on a function call, you jump into the implementation).

Take a look at this video: https://youtu.be/7EmboKQH8lM